### ISSUE TITLE
test_stl and test_mstl do not find test results

### ISSUE THREAD

**Issue by bnavigator** [statsmodels/statsmodels#8928]
#### Describe the bug

When running pytest from a non-in-place installation `test_stl` fails at collection time and `test_mstl` at test time. (We do this in order to check the openSUSE rpm package, so we do not want to build in-place).

#### Code Sample, a copy-pastable example if possible

```shell
mkdir testdir
pushd testdir
# use the pytest config but discover tests from installed packagte
cp $statsmodelssrc/setup.cfg .
python3 -m venv myvenv
myvenv/bin/pip install 'statsmodels[develop]'
myvenv/bin/pytest -v myvenv/lib/python*/site-packages/statsmodels
myvenv/bin/pytest -v -x -k "test_mstl" myvenv/lib/python*/site-packages/statsmodels --ignore myvenv/lib/python*/site-packages/statsmodels/tsa/stl/tests/test_stl.py
```

#### Output

<details>

```
ben@skylab:~/tmp/testdir]% myvenv/bin/pytest -v myvenv/lib/python*/site-packages/statsmodels                         [1]
================================================== test session starts ===================================================
platform linux -- Python 3.11.3, pytest-7.0.1, pluggy-1.2.0 -- /home/ben/tmp/testdir/myvenv/bin/python3
cachedir: .pytest_cache
Using --randomly-seed=4162255313
rootdir: /home/ben/tmp/testdir/myvenv/lib/python3.11/site-packages/statsmodels, configfile: setup.cfg
plugins: xdist-3.3.1, randomly-3.12.0
collected 18026 items / 1 error / 18025 selected                                                                         

========================================================= ERRORS =========================================================
_______________________________________ ERROR collecting tsa/stl/tests/test_stl.py _______________________________________
/usr/lib64/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1206: in _gcd_import
    ???
<frozen importlib._bootstrap>:1178: in _find_and_load
    ???
<frozen importlib._bootstrap>:1149: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:690: in _load_unlocked
    ???
myvenv/lib64/python3.11/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
myvenv/lib64/python3.11/site-packages/statsmodels/tsa/stl/tests/test_stl.py:13: in <module>
    results = pd.read_csv(file_path)
myvenv/lib64/python3.11/site-packages/pandas/io/parsers/readers.py:912: in read_csv
    return _read(filepath_or_buffer, kwds)
myvenv/lib64/python3.11/site-packages/pandas/io/parsers/readers.py:577: in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
myvenv/lib64/python3.11/site-packages/pandas/io/parsers/readers.py:1407: in __init__
    self._engine = self._make_engine(f, self.engine)
myvenv/lib64/python3.11/site-packages/pandas/io/parsers/readers.py:1661: in _make_engine
    self.handles = get_handle(
myvenv/lib64/python3.11/site-packages/pandas/io/common.py:859: in get_handle
    handle = open(
E   FileNotFoundError: [Errno 2] No such file or directory: '/home/ben/tmp/testdir/myvenv/lib64/python3.11/site-packages/statsmodels/tsa/stl/tests/results/stl_test_results.csv'
================================================ short test summary info =================================================
ERROR myvenv/lib/python3.11/site-packages/statsmodels/tsa/stl/tests/test_stl.py - FileNotFoundError: [Errno 2] No such ...
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
==================================================== 1 error in 4.19s ====================================================
```

```
[ben@skylab:~/tmp/testdir]% myvenv/bin/pytest -v -x -k "test_mstl" myvenv/lib/python*/site-packages/statsmodels --ignore myvenv/lib/python*/site-packages/statsmodels/tsa/stl/tests/test_stl.py
================================================== test session starts ===================================================
platform linux -- Python 3.11.3, pytest-7.0.1, pluggy-1.2.0 -- /home/ben/tmp/testdir/myvenv/bin/python3
cachedir: .pytest_cache
Using --randomly-seed=3298185568
rootdir: /home/ben/tmp/testdir/myvenv/lib/python3.11/site-packages/statsmodels, configfile: setup.cfg
plugins: xdist-3.3.1, randomly-3.12.0
collected 18026 items / 18008 deselected / 18 selected                                                                   

myvenv/lib/python3.11/site-packages/statsmodels/tsa/stl/tests/test_mstl.py::test_seasonal_is_datafame_when_input_pandas_and_multiple_periods ERROR [  5%]

========================================================= ERRORS =========================================================
___________________ ERROR at setup of test_seasonal_is_datafame_when_input_pandas_and_multiple_periods ___________________

    @pytest.fixture(scope="function")
    def data_pd():
        cur_dir = Path(__file__).parent.resolve()
        file_path = cur_dir / "results/mstl_elec_vic.csv"
>       return pd.read_csv(file_path, index_col=["ds"], parse_dates=["ds"])

myvenv/lib64/python3.11/site-packages/statsmodels/tsa/stl/tests/test_mstl.py:21: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
myvenv/lib64/python3.11/site-packages/pandas/io/parsers/readers.py:912: in read_csv
    return _read(filepath_or_buffer, kwds)
myvenv/lib64/python3.11/site-packages/pandas/io/parsers/readers.py:577: in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
myvenv/lib64/python3.11/site-packages/pandas/io/parsers/readers.py:1407: in __init__
    self._engine = self._make_engine(f, self.engine)
myvenv/lib64/python3.11/site-packages/pandas/io/parsers/readers.py:1661: in _make_engine
    self.handles = get_handle(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

path_or_buf = PosixPath('/home/ben/tmp/testdir/myvenv/lib/python3.11/site-packages/statsmodels/tsa/stl/tests/results/mstl_elec_vic.csv')
mode = 'r'

    @doc(compression_options=_shared_docs["compression_options"] % "path_or_buf")
    def get_handle(
        path_or_buf: FilePath | BaseBuffer,
        mode: str,
        *,
        encoding: str | None = None,
        compression: CompressionOptions = None,
        memory_map: bool = False,
        is_text: bool = True,
        errors: str | None = None,
        storage_options: StorageOptions = None,
    ) -> IOHandles[str] | IOHandles[bytes]:
        """
        Get file handle for given path/buffer and mode.
    
        Parameters
        ----------
        path_or_buf : str or file handle
            File path or object.
        mode : str
            Mode to open path_or_buf with.
        encoding : str or None
            Encoding to use.
        {compression_options}
    
            .. versionchanged:: 1.0.0
               May now be a dict with key 'method' as compression mode
               and other keys as compression options if compression
               mode is 'zip'.
    
            .. versionchanged:: 1.1.0
               Passing compression options as keys in dict is now
               supported for compression modes 'gzip', 'bz2', 'zstd' and 'zip'.
    
            .. versionchanged:: 1.4.0 Zstandard support.
    
        memory_map : bool, default False
            See parsers._parser_params for more information. Only used by read_csv.
        is_text : bool, default True
            Whether the type of the content passed to the file/buffer is string or
            bytes. This is not the same as `"b" not in mode`. If a string content is
            passed to a binary file/buffer, a wrapper is inserted.
        errors : str, default 'strict'
            Specifies how encoding and decoding errors are to be handled.
            See the errors argument for :func:`open` for a full list
            of options.
        storage_options: StorageOptions = None
            Passed to _get_filepath_or_buffer
    
        .. versionchanged:: 1.2.0
    
        Returns the dataclass IOHandles
        """
        # Windows does not default to utf-8. Set to utf-8 for a consistent behavior
        encoding = encoding or "utf-8"
    
        errors = errors or "strict"
    
        # read_csv does not know whether the buffer is opened in binary/text mode
        if _is_binary_mode(path_or_buf, mode) and "b" not in mode:
            mode += "b"
    
        # validate encoding and errors
        codecs.lookup(encoding)
        if isinstance(errors, str):
            codecs.lookup_error(errors)
    
        # open URLs
        ioargs = _get_filepath_or_buffer(
            path_or_buf,
            encoding=encoding,
            compression=compression,
            mode=mode,
            storage_options=storage_options,
        )
    
        handle = ioargs.filepath_or_buffer
        handles: list[BaseBuffer]
    
        # memory mapping needs to be the first step
        # only used for read_csv
        handle, memory_map, handles = _maybe_memory_map(handle, memory_map)
    
        is_path = isinstance(handle, str)
        compression_args = dict(ioargs.compression)
        compression = compression_args.pop("method")
    
        # Only for write methods
        if "r" not in mode and is_path:
            check_parent_directory(str(handle))
    
        if compression:
            if compression != "zstd":
                # compression libraries do not like an explicit text-mode
                ioargs.mode = ioargs.mode.replace("t", "")
            elif compression == "zstd" and "b" not in ioargs.mode:
                # python-zstandard defaults to text mode, but we always expect
                # compression libraries to use binary mode.
                ioargs.mode += "b"
    
            # GZ Compression
            if compression == "gzip":
                if isinstance(handle, str):
                    # error: Incompatible types in assignment (expression has type
                    # "GzipFile", variable has type "Union[str, BaseBuffer]")
                    handle = gzip.GzipFile(  # type: ignore[assignment]
                        filename=handle,
                        mode=ioargs.mode,
                        **compression_args,
                    )
                else:
                    handle = gzip.GzipFile(
                        # No overload variant of "GzipFile" matches argument types
                        # "Union[str, BaseBuffer]", "str", "Dict[str, Any]"
                        fileobj=handle,  # type: ignore[call-overload]
                        mode=ioargs.mode,
                        **compression_args,
                    )
    
            # BZ Compression
            elif compression == "bz2":
                # Overload of "BZ2File" to handle pickle protocol 5
                # "Union[str, BaseBuffer]", "str", "Dict[str, Any]"
                handle = _BZ2File(  # type: ignore[call-overload]
                    handle,
                    mode=ioargs.mode,
                    **compression_args,
                )
    
            # ZIP Compression
            elif compression == "zip":
                # error: Argument 1 to "_BytesZipFile" has incompatible type
                # "Union[str, BaseBuffer]"; expected "Union[Union[str, PathLike[str]],
                # ReadBuffer[bytes], WriteBuffer[bytes]]"
                handle = _BytesZipFile(
                    handle, ioargs.mode, **compression_args  # type: ignore[arg-type]
                )
                if handle.buffer.mode == "r":
                    handles.append(handle)
                    zip_names = handle.buffer.namelist()
                    if len(zip_names) == 1:
                        handle = handle.buffer.open(zip_names.pop())
                    elif not zip_names:
                        raise ValueError(f"Zero files found in ZIP file {path_or_buf}")
                    else:
                        raise ValueError(
                            "Multiple files found in ZIP file. "
                            f"Only one file per ZIP: {zip_names}"
                        )
    
            # TAR Encoding
            elif compression == "tar":
                compression_args.setdefault("mode", ioargs.mode)
                if isinstance(handle, str):
                    handle = _BytesTarFile(name=handle, **compression_args)
                else:
                    # error: Argument "fileobj" to "_BytesTarFile" has incompatible
                    # type "BaseBuffer"; expected "Union[ReadBuffer[bytes],
                    # WriteBuffer[bytes], None]"
                    handle = _BytesTarFile(
                        fileobj=handle, **compression_args  # type: ignore[arg-type]
                    )
                assert isinstance(handle, _BytesTarFile)
                if "r" in handle.buffer.mode:
                    handles.append(handle)
                    files = handle.buffer.getnames()
                    if len(files) == 1:
                        file = handle.buffer.extractfile(files[0])
                        assert file is not None
                        handle = file
                    elif not files:
                        raise ValueError(f"Zero files found in TAR archive {path_or_buf}")
                    else:
                        raise ValueError(
                            "Multiple files found in TAR archive. "
                            f"Only one file per TAR archive: {files}"
                        )
    
            # XZ Compression
            elif compression == "xz":
                # error: Argument 1 to "LZMAFile" has incompatible type "Union[str,
                # BaseBuffer]"; expected "Optional[Union[Union[str, bytes, PathLike[str],
                # PathLike[bytes]], IO[bytes]]]"
                handle = get_lzma_file()(handle, ioargs.mode)  # type: ignore[arg-type]
    
            # Zstd Compression
            elif compression == "zstd":
                zstd = import_optional_dependency("zstandard")
                if "r" in ioargs.mode:
                    open_args = {"dctx": zstd.ZstdDecompressor(**compression_args)}
                else:
                    open_args = {"cctx": zstd.ZstdCompressor(**compression_args)}
                handle = zstd.open(
                    handle,
                    mode=ioargs.mode,
                    **open_args,
                )
    
            # Unrecognized Compression
            else:
                msg = f"Unrecognized compression type: {compression}"
                raise ValueError(msg)
    
            assert not isinstance(handle, str)
            handles.append(handle)
    
        elif isinstance(handle, str):
            # Check whether the filename is to be opened in binary mode.
            # Binary mode does not support 'encoding' and 'newline'.
            if ioargs.encoding and "b" not in ioargs.mode:
                # Encoding
>               handle = open(
                    handle,
                    ioargs.mode,
                    encoding=ioargs.encoding,
                    errors=errors,
                    newline="",
                )
E               FileNotFoundError: [Errno 2] No such file or directory: '/home/ben/tmp/testdir/myvenv/lib/python3.11/site-packages/statsmodels/tsa/stl/tests/results/mstl_elec_vic.csv'

myvenv/lib64/python3.11/site-packages/pandas/io/common.py:859: FileNotFoundError
================================================ short test summary info =================================================
ERROR myvenv/lib/python3.11/site-packages/statsmodels/tsa/stl/tests/test_mstl.py::test_seasonal_is_datafame_when_input_pandas_and_multiple_periods
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=========================================== 18008 deselected, 1 error in 4.16s ==========================================
```

</details>

#### Expected Output

Successful test suite

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>

```

INSTALLED VERSIONS
------------------
Python: 3.11.3.final.0
OS: Linux 6.3.7-1-default #1 SMP PREEMPT_DYNAMIC Mon Jun 12 05:01:32 UTC 2023 (b5f9ff5) x86_64
byteorder: little
LC_ALL: None
LANG: de_DE.UTF-8

statsmodels
===========

Installed: 0.14.0 (/home/ben/tmp/testdir/myvenv/lib64/python3.11/site-packages/statsmodels)

Required Dependencies
=====================

cython: 0.29.35 (/home/ben/tmp/testdir/myvenv/lib64/python3.11/site-packages/Cython)
numpy: 1.23.2 (/home/ben/tmp/testdir/myvenv/lib64/python3.11/site-packages/numpy)
scipy: 1.10.1 (/home/ben/tmp/testdir/myvenv/lib64/python3.11/site-packages/scipy)
pandas: 2.0.2 (/home/ben/tmp/testdir/myvenv/lib64/python3.11/site-packages/pandas)
    dateutil: 2.8.2 (/home/ben/tmp/testdir/myvenv/lib64/python3.11/site-packages/dateutil)
patsy: 0.5.3 (/home/ben/tmp/testdir/myvenv/lib64/python3.11/site-packages/patsy)

Optional Dependencies
=====================

matplotlib: 3.7.1 (/home/ben/tmp/testdir/myvenv/lib64/python3.11/site-packages/matplotlib)
    backend: TkAgg 
cvxopt: Not installed
joblib: 1.2.0 (/home/ben/tmp/testdir/myvenv/lib64/python3.11/site-packages/joblib)

Developer Tools
================

IPython: Not installed
    jinja2: Not installed
sphinx: Not installed
    pygments: Not installed
pytest: 7.0.1 (/home/ben/tmp/testdir/myvenv/lib64/python3.11/site-packages/pytest)
virtualenv: Not installed
```

</details>

Please add the missing result files to the setuptools data or consider removing all result files from the installed package (published wheel) and move to a separate directory/repository.

**Comment by bashtage** [comment]
I think there is an issue already about this.

**Comment by bnavigator** [comment]
I didn't find it

**Comment by bashtage** [comment]
Looks like there is no issue here, at least in master. There is only 1 supported way to test the installed package, using ` test()` command.

```
python -c "import statsmodels.tsa.stl as stl; stl.test()"
```

which returns

```
Running pytest C:\Anaconda\lib\site-packages\statsmodels\tsa\stl --tb=short --disable-pytest-warnings
================================================================================================================================================= test session starts =================================================================================================================================================
platform win32 -- Python 3.10.11, pytest-7.4.0, pluggy-1.0.0
rootdir: C:\Anaconda\lib\site-packages\statsmodels
configfile: setup.cfg
plugins: anyio-3.7.0, cov-4.0.0, xdist-3.3.1
collected 37 items                                                                                                                                                                                                                                                                                                     

..\Anaconda\lib\site-packages\statsmodels\tsa\stl\tests\test_mstl.py ..................                                                                                                                                                                                                                          [ 48%]
..\Anaconda\lib\site-packages\statsmodels\tsa\stl\tests\test_stl.py ...................                                                                                                                                                                                                                          [100%]

=========================================================================================================================================== 37 passed, 2 warnings in 14.65s =========================================================================================================================================== 
```

**Comment by bashtage** [comment]
Was fixed in  081c9ac in master. No point release since then.

**Comment by bnavigator** [comment]
Sorry for the noise. Apparently the GitHub search bar failed for me when I searched for various strings of the error message and file names.
